### PR TITLE
Single logout Support

### PR DESCRIPTION
**Pull request by Winitzky**
release notes:
- Added logoutresponse.go to handle the logout respone (like authresponse)
- Added Sign.go to handle signature of logout request (needed to sign a string and not an xml)
- Added fields to EntityDescriptor to provide a valid updated metadata
- Added fields to SPSSODescriptor to match 
- Added 2 types - logoutresponse and relaystate for better handling these cases
- Modified logoutrequest.go to match http-redirect

and thats about it - tested it on 3 scenarios:
1. SSO - didn't break backwards compatability
2. SLO - Achieved
3. Metadata - provides a valid metadata file with logout service included

### CODE REVIEW

File: logoutresponse.go [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+	err := xml.Unmarshal(responseXML, &response)
+	if err != nil {
+		return nil, err
+	}
+	// save the original response because XML Signatures are fussy
+	response.originalString = string(responseXML)
+	return &response, nil
+}
+
+func (r *LogoutResponse) Validate(s *ServiceProviderSettings) error {
```
Comment (dorsha) [inline logoutresponse.go:50]: We will need to check the validation with serval Identify Providers (ADFS, PingFederate, ..)

File: util/Sign.go [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+	}
+
+
+	signed, err := signer.Sign([]byte(toSign))
+	if err != nil {
+		fmt.Errorf("could not sign request: %v", err)
+		return "", err
+	}
+	sig := base64.StdEncoding.EncodeToString(signed)
+	fmt.Printf("Signature: %v\n", sig)
```
Comment (dorsha) [inline util/Sign.go]: remove print

**Comment by dorsha** [comment]
Nice, small comments.

File: util/Sign.go (hunk 2 above)
Comment (Winitzky) [reply util/Sign.go]: Removed

File: logoutresponse.go (hunk 1 above)
Comment (Winitzky) [reply logoutresponse.go:50]: Agreed

**Pull request merged**